lib/ui/exec.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const blessed = require('blessed'),
+      os      = require('os');
+
+class Exec {
+
+  constructor(screen, namespace, pod, container, user) {
+    let terminal, term, self = this;
+
+    this.write = function (data) {
+      const payload     = Buffer.allocUnsafe(data.length + 1);
+      // send to STDIN
+      payload.writeUInt8(0, 0);
+      if(typeof data === "string") {
+        // browser
+        payload.write(data, 1, 'binary');
+      } else {
+        data.copy(payload, 1);
+      }
+      self.sendData(payload);
+    }
+
+    this.kill = function (data) {
+      screen.grabKeys = false;
+      terminal.kill();
+    }
+
+    this.sendResize = function () {
+      if(!self.sendData) {
+        return;
+      }
+      const adjust = '{"Width":' + term.cols + ',"Height":' + term.rows + '}';
+      const length = Buffer.byteLength(adjust)
+      const buffer = Buffer.allocUnsafe(length + 1);
+      buffer.writeUInt8(4, 0);
+      buffer.write(adjust, 1, 'binary');
+      self.sendData(buffer);
+    }
+
+    // TODO: use xterm.js when in browser
```

Comment:
We can try using https://github.com/rse/blessed-xterm. If that works though, I think it may be better to duplicate the component and remove the dependency on `node-pty`.